k8s/k8s.go

```diff
@@ -294,6 +296,45 @@ func CmdAddK8s(args *skel.CmdArgs, conf utils.NetConf, nodename string, calicoCl
 	return result, nil
 }
 
+// ContainerIDMismatchErr is for when ContainerID does not match ActiveInstanceID during Deletion process.
+var ContainerIDMismatchErr = errors.New("ContainerID does not match ActiveInstanceID")
+
+// CmdDelK8s performs some special checks for the "DEL" operation on a kubernetes pod.
+// The following logic only applies to kubernetes since it sends multiple DELs for the same endpoint.
+// We store CNI_CONTAINERID as ActiveInstanceID in WEP Metadata for k8s,
+// so we need to make sure we check if ContainerID and ActiveInstanceID are the same before deleting the pod.
+func CmdDelK8s(c *calicoclient.Client, ep api.WorkloadEndpointMetadata, args *skel.CmdArgs, logger *log.Entry) (*api.WorkloadEndpointMetadata, error) {
+
+	wep, err := c.WorkloadEndpoints().Get(ep)
+	if err != nil {
+		if _, ok := err.(cerrors.ErrorResourceDoesNotExist); ok {
+			// We can talk to the datastore but WEP doesn't exist in there,
+			// but we still want to go ahead with the clean up. So, log a warning and clean up.
+			logger.WithField("WorkloadEndpoint", ep).Warning("WorkloadEndpoint does not exist in the datastore, moving forward with the clean up")
```

Comment:
I _think_ you mean to return nil, nil here (hence your comment below).  If that is the case then:

You can lose the next `else` and you can drop the `if wep != nil {` below and unindent that whole section.